bson/__init__.py

```diff
@@ -999,6 +999,17 @@ def _decode_selective(rawdoc, fields, codec_options):
     return doc
 
 
+def _convert_raw_document_lists_to_streams(document, fields):
+    for key, value in fields.items():
+        if key in document:
+            if value == 1:
+                if isinstance(document[key], list) and len(document[key]) > 0:
+                    stream = b"".join(doc.raw for doc in document[key])
+                    document[key] = [stream]
+            else:
+                _convert_raw_document_lists_to_streams(document[key], value)
```

Comment:
Can we make this code _less_ generic so that it is more readable? Something like this would be a lot easier to follow and maintain:
```
cursor = document.get('cursor')
if not cursor:
    return
for key in ('firstBatch', 'nextBatch'):
    batch = cursor.get(key)
    if batch is None:
        continue 
    cursor[key] = b"".join(doc.raw for doc in batch)
```

Reply:
Done.